### PR TITLE
21597 - Query build card save modal errors

### LINKED ISSUE #21597: Query Build Card Save modal does not properly show errors from API
I ran into this while fixing #14145.

After PR #21598, the API returns a 400 error response (actually 202 with `_status` since the endpoint is async, but still effectively a 400):

```json
{"message":"Invalid Field Filter: Field 164574 \"PRODUCTS\".\"CATEGORY\" belongs to Database 2276 \"sample-dataset\", but the query is against Database 2275 \"test-data\"","_status":400}
```

But the Save Card modal just shows a Type Error:

image

Console logs:

```
Form.jsx:216 Form submission error TypeError: Cannot read properties of undefined (reading 'collection')
    at reducer (schemas.js:56:1)
    at redux.js:163:1
    at combination (combineReducers.js:120:1)
    at combination (combineReducers.js:120:1)
    at dispatch (createStore.js:165:1)
    at middleware.js:22:1
    at index.js:28:1
    at store.js:25:1
    at dispatch (applyMiddleware.js:35:1)
    at _callee3$ (redux.js:210:1)
```

It should show the actual error message here.

Note that #14145 is a custom success escalation so this should probably be considered one as well

### PR DESCRIPTION
**Pull request by losrebellos**
Fixes #21597.

__After__

image

### CODE REVIEW

File: frontend/src/metabase/containers/Form.jsx [hunk 1]
```diff
@@ -232,7 +232,11 @@ export default class Form extends React.Component {
         };
       } else if (error) {
         throw {
-          _error: error.data.message || error.data,
+          _error:
```
Comment (losrebellos) [inline frontend/src/metabase/containers/Form.jsx:235]: Error order updated:
1. `message` from `error.data`
2. `Error.message`
3. `error.data` (might be a string?)
4. default error message

File: frontend/src/metabase/entities/schemas.js [hunk 2]
```diff
@@ -52,15 +52,23 @@ export default createEntity({
 
   reducer: (state = {}, { type, payload }) => {
     if (type === Questions.actionTypes.CREATE) {
-      const { question } = payload;
-      const schema = getCollectionVirtualSchemaId(question.collection);
-      if (!state[schema]) {
-        return state;
+      const { question, status, data } = payload;
+      if (question) {
```
Comment (losrebellos) [inline frontend/src/metabase/entities/schemas.js:56]: Only if `question` exists

File: frontend/src/metabase/entities/schemas.js [hunk 3]
```diff
@@ -52,15 +52,23 @@ export default createEntity({
...
+        }
+        const virtualQuestionId = getQuestionVirtualTableId(question);
+        return updateIn(state, [schema, "tables"], tables =>
+          addTableAvoidingDuplicates(tables, virtualQuestionId),
+        );
+      }
+      // IF there is no question
+      // AND if the request has failed,
+      // throw the error message to display
+      else if (status === 400 && data?.message) {
```
Comment (losrebellos) [inline frontend/src/metabase/entities/schemas.js:69]: Throw an error if the API has an error

File: frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.jsx [hunk 4]
```diff
@@ -168,8 +168,11 @@ export default class TagEditorParam extends Component {
                 {() => (
                   <SchemaTableAndFieldDataSelector
                     databases={databases}
-                    selectedDatabaseId={database ? database.id : null}
-                    selectedTableId={table ? table.id : null}
+                    selectedDatabase={database || null}
+                    selectedDatabaseId={database?.id || null}
+                    selectedTable={table || null}
+                    selectedTableId={table?.id || null}
+                    selectedField={field || null}
```
Comment (losrebellos) [inline frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.jsx:175]: Pass the missing properties `selectedDatabase`, `selectedTable` and `selectedField` so on the first page/app load, the sidebar is rendered properly.

**Review by flamber — APPROVED** [review]
This works great.

**Comment by ariya** [comment]
Is it possible to write a simple repro for this?

**Review by camsaul — APPROVED** [review]

**Comment by ariya** [comment]
> Repro created by it would be great to have @nemanjaglumac's feedback on that part of the code as I am sure there is some cypress `utils` that I could have used...

Repro looks solid! We can always polish and tweak it afterwards.

**Review by ariya — COMMENTED** [review]
Go for it :rocket:

**Pull request merged**